packages/api/src/dao/BlocksDAO.js

```diff
@@ -48,6 +48,53 @@ module.exports = class BlockDAO {
     return Block.fromRow({ header: block, txs })
   }
 
+  getBlockByValidator = async (validator, page, limit, order) => {
+    const fromRank = ((page - 1) * limit) + 1
+    const toRank = fromRank + limit - 1
+
+    const subquery = this.knex('blocks')
+      .select(
+        'blocks.hash as hash',
+        'blocks.height as height',
+        'blocks.timestamp as timestamp',
+        'blocks.block_version as block_version',
+        'blocks.app_version as app_version',
+        'blocks.l1_locked_height as l1_locked_height',
+        'blocks.validator as validator',
+        this.knex.raw(`rank() over (partition by blocks.validator order by blocks.height ${order}) as rank`)
+      )
+      .where('blocks.validator', validator)
+      .as('blocks')
+
+    const rows = await this.knex(subquery)
+      .select(this.knex('blocks').count('height').as('total_count').where('blocks.validator', validator),
+        'blocks.hash as hash', 'height', 'timestamp', 'block_version',
+        'app_version', 'l1_locked_height', 'state_transitions.hash as st_hash', 'validator')
+      .whereBetween('rank', [fromRank, toRank])
+      .orderBy('blocks.height', order)
+      .leftJoin('state_transitions', 'state_transitions.block_hash', 'blocks.hash')
+
+    const totalCount = rows.length > 0 ? Number(rows[0].total_count) : 0
+
+    const blocksMap = rows.reduce((blocks, row) => {
+      const block = blocks[row.hash]
+      const { st_hash: txHash } = row
```

Comment:
There is no `st_hash` in the selected fields, therefore this reducer does not sum up transactions